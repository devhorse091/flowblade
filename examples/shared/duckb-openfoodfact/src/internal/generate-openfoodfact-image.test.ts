import { generateOpenfoodfactImage } from './generate-openfoodfact-image';

describe('generate-openfoodfact-image', () => {
  it('should return image information', () => {
    const image = generateOpenfoodfactImage({
      code: '3068320123264',
      images: [
        {
          key: '1',
          imgid: 99,
          sizes: {
            '100': { h: 100, w: 100 },
            '400': { h: 400, w: 100 },
          },
        },
        {
          key: 'front_fr',
          imgid: 100,
          sizes: {
            '100': { h: 100, w: 100 },
            '400': { h: 400, w: 100 },
          },
        },
      ],
    });
    expect(image).toMatchSnapshot();
  });
});
