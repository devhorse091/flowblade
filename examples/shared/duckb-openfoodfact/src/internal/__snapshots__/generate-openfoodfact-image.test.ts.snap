// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generate-openfoodfact-image > should return image information 1`] = `
{
  "baseUrl": "https://images.openfoodfacts.org/images/products",
  "image": {
    "fileName": "front_fr.100.400.jpg",
    "imgid": 100,
    "key": "front_fr",
    "resolutions": [
      100,
      400,
    ],
    "url": "https://images.openfoodfacts.org/images/products/306/832/012/3264/1.400.jpg",
  },
  "prefix": "306/832/012/3264",
}
`;
